src/Utils.vala

```diff
@@ -55,6 +55,37 @@ namespace Utils {
         return usermanager;
     }
 
+    private static Polkit.Permission? permission = null;
+
+    public static Polkit.Permission? get_permission () {
+        if (permission != null)
+            return permission;
+        try {
+            permission = new Polkit.Permission.sync ("io.elementary.initial-setup", new Polkit.UnixProcess (Posix.getpid ()));
+            return permission;
+        } catch (Error e) {
+            critical (e.message);
+            return null;
+        }
+    }
+
+    public static void create_new_user (string fullname, string username, string password) {
+        if (get_permission ().allowed) {
```

Comment:
get_permission () signature indicates a null return is possible so this should be tested for.